nntrainer/utils/common_properties.h

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file common_properties.h
+ * @date 09 April 2021
+ * @brief This file contains list of common properties widely used across layers
+ * @see	https://github.com/nnstreamer/nntrainer
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+
+#include <base_properties.h>
+
+#ifndef __COMMON_PROPERTIES_H__
+#define __COMMON_PROPERTIES_H__
+
+namespace nntrainer {
+namespace props {
+
+/**
+ * @brief unit property, unit is used to measure how many weights are there
+ *
+ */
+class Unit : public nntrainer::Property<unsigned int> {
```

Comment:
you might consider making a class `PositiveIntegers`, where defaults to 0, `is_valid` is `value > 0`, and `prop_tag` is `uint_prop_tag`, as this will common for a lot of properties. Might come in handy.

Reply:
Sounds good, I'll make it as a separate patch onward

Reply:
sure.